src/vs/editor/contrib/snippet/snippetVariables.ts

```diff
@@ -272,13 +274,31 @@ export class WorkspaceBasedVariableResolver implements VariableResolver {
 		}
 
 		if (isSingleFolderWorkspaceIdentifier(workspaceIdentifier)) {
-			return path.basename(workspaceIdentifier.path);
+			if (variable.name === 'WORKSPACE_NAME') {
```

Comment:
this repeated-if-else is a little ugly, can't you change it be two methods, like `_resolveWorkspaceName` and `_resoveWorkspacePath`?